packages/salesforcedx-vscode-core/src/orgBrowser/metadataType.ts

```diff
@@ -138,31 +81,27 @@ export type MetadataObject = {
   xmlName: string;
 };
 
-export function buildTypesList(metadataTypesPath: string): string[] {
-  try {
-    const fileData = JSON.parse(fs.readFileSync(metadataTypesPath, 'utf8'));
-    const metadataObjects = fileData.metadataObjects as MetadataObject[];
-    const metadataTypes = [];
-    for (const metadataObject of metadataObjects) {
-      if (!isNullOrUndefined(metadataObject.xmlName)) {
-        metadataTypes.push(metadataObject.xmlName);
-      }
+export function buildTypesList(
+  metadataFile?: any,
+  metadataTypesPath?: string
+): string[] {
+  if (isNullOrUndefined(metadataFile)) {
+    try {
+      metadataFile = fs.readFileSync(metadataTypesPath!, 'utf8');
+    } catch (e) {
+      throw e;
     }
-    telemetryService.sendEventData('Metadata Types Quantity', undefined, {
-      metadataTypes: metadataTypes.length
-    });
-    return metadataTypes;
-  } catch (e) {
-    throw e;
   }
-}
-
-export async function onUsernameChange() {
-  const metadataTypesPath = await getTypesPath();
-  if (
-    !isNullOrUndefined(metadataTypesPath) &&
-    !fs.existsSync(metadataTypesPath)
-  ) {
-    await forceDescribeMetadata(metadataTypesPath);
+  const jsonObject = JSON.parse(metadataFile);
```

Comment:
Wasn't sure if I should wrap the JSON parsing here and for the components in a try/catch

Reply:
I think we should expand the try/catch to cover the rest of the code since it's also navigating a json structure that might throw a null/empty value.